### ISSUE TITLE
Error parsing file when CR appears in header in quotes

### ISSUE THREAD

**Issue by lboynton** [mholt/PapaParse#252]
Hi Matt, thanks for the awesome library!

I have encountered a potential issue regarding newline characters, and it would be great to hear your thoughts on it. 

Some files I'm using rather unusually have CR characters in the headers, although they appear in quotes. The end of the line is a normal LF character. I think when PapaParse tries to automatically detect the newline characters, it will use CR which leads to parsing errors (MissingQuotes). Obviously I could tell PapaParse the newline character, but ideally I'd like to continue using the automatic detection because I don't always know what it's going to be.

Would it be possible to ignore newline characters which are in quotes when trying to guess the CSV end of line character?

Here is an example file which exhibits the problem: https://www.dropbox.com/s/d6qkvywinjekulh/header-newline.csv?dl=0

Thanks!

**Comment by lboynton** [comment]
I've added a failing test here if it helps: https://github.com/lboynton/PapaParse/commit/98efe3063fbcd275a074c38d9358674b2e38e399

**Comment by mholt** [comment]
Hi Lee,

Thanks for opening the issue. I haven't had much time to look at it in detail yet as I'm at a conference today, but I can look into it over the next few weeks to see if I can understand the behavior you're describing. Thanks for providing an example file; I'll look soon.

**Comment by lboynton** [comment]
No problem, thanks for responding so quickly :)

**Comment by mholt** [comment]
This is how we currently detect line endings: https://github.com/mholt/PapaParse/blob/9183ffb2d815bfe22aca0c65782b8b0eeeb07d75/papaparse.js#L944-L961

1. If there are no `\r` in the first 1 MB of the file, we assume `\n`.
2. Otherwise, we count how many `\n` appear immediately following an `\r`
3. If there are close to the number of `\n`s as `\r`s, assume `\r\n`, otherwise assume `\r`.

So what's happening with your input is that the `\r` is confusing the parser because there are so few `\n` characters (very few rows). But add a few more rows without `\r` and the parser won't get confused. (Try it!)

Is your data really frequently as short as 2-3 rows? If your carriage return characters only appear in the header then the parser should be able to guess correctly with at least a few rows.

**Comment by lboynton** [comment]
Thanks for having a look into my issue Matt. I think my problem though is that my file uses `\n` line endings which in the list above is never an outcome of the line ending detection if there is a `\r` in the header?

My data is rarely that short btw, I was trying to provide the smallest possible test case.

**Comment by mholt** [comment]
I think I see what you mean now. This should get fixed.

**Comment by dpatte** [comment]
I seem to have a similar problem...
some of my quoted string fields may contain either of the newline characters (carriage return or line feed). I was expecting these charcters to be assumed to be part of the string if found within quoted strings, but papaparse is interpretting them as record newlines. I am using auto newlines.

Note that when I originally write the csv files using php, php sees these characters as parts of strings, and automatically encloses these strings in quotes.

**Comment by thj-dk** [comment]
I'm having the same issues as @dpatte. Did you ever get this solved?

**Comment by mholt** [comment]
No, I've been too busy with other projects lately, but pull requests are welcome! In fact, I wonder if #281 would have any effect on this issue.

**Comment by dpatte** [comment]
I solved it by prescanning my input and converting CR and LF to character strings "<CR>" and "<LF>" , parsing, then restoring them to CR LF after the parse. Worked for me.

**Comment by pokoli** [comment]
Hi All, 

I have tryied to solve the problem by replacing all the text inside quotes by an empty string as it's not relevant for detecting the line delimiter. See https://github.com/mholt/PapaParse/compare/master...newline-problem

But it seems to not solve the test case. Any help will be much appreciated. 

P.D: I refactored a litle bit the test case (See: https://github.com/mholt/PapaParse/commit/a3e3e6dc94201b255a6dd462feea2d09982b3f59) as it was an extra comment and I think that skipEmpty lines is not requied.
